### PR TITLE
Svelte: Paper cuts for 5.5 release

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://linear.app/sourcegraph/issue/SRCH-78/svelte-improve-repository-revision-picker-ui

Just a package of small fixes in different places (group them in one PR to avoid having a lot of super small PRs before the release. These fixes include

## Add Search files quick actions for collapsed file sidebar state
Screenshot 2024-05-31 at 15 34 12

## Updates for tabs appearance 
Screenshot 2024-05-31 at 15 34 36

## Update fuzzy finder  appearance (most of the styles that we have in the react version and fixes for Safari)
Screenshot 2024-05-31 at 15 35 42

- Add keyboard support for tabs 
- Fixes opt-in/out tooltips delays
- Add close-on escape feature to fuzzy finder and popover components
- Fix sidebar filters panel shadow problem 
- Add shortcuts to repo revision popover tabs and history and references tabs

## Test plan
- Manual testing of all changes listed above

### CODE REVIEW

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte [hunk 1]
```diff
@@ -24,10 +26,20 @@
             return
         }
     }
+
+    const historyHotkey: Keys = {
+        key: 'alt+h',
+        mac: '⌥+h',
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte:32]: You only need to set `mac` if you want to have a different keyboard shortcut. For displaying purposes, `alt` should be rendered as `⌥` on macOS. If it doesn't we need to update our rendering logic.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte [hunk 2]
```diff
@@ -24,10 +26,20 @@
...
+    }
+
+    const referenceHotkey: Keys = {
+        key: 'alt+r',
+        mac: '⌥+r',
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte:37]: Same as above.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte [hunk 3]
```diff
@@ -9,6 +10,21 @@
 
     export type RepositoryCommits = { nodes: RevPickerGitCommit[] }
     export type RepositoryGitCommit = RevPickerGitCommit
+
+    const branchesHotkey: Keys = {
+        key: 'shift+b',
+        mac: 'shift+b',
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte:16]: No need to specify separate mac keyboard shortcut.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte [hunk 4]
```diff
@@ -9,6 +10,21 @@
...
+    }
+
+    const tagsHotkey: Keys = {
+        key: 'shift+t',
+        mac: 'shift+t',
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte:21]: No need to specify separate mac keyboard shortcut.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte [hunk 5]
```diff
@@ -9,6 +10,21 @@
...
+    }
+
+    const commitsHotkey: Keys = {
+        key: 'shift+c',
+        mac: 'shift+c',
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte:26]: No need to specify separate mac keyboard shortcut.

**Review by fkling — APPROVED** [review]
👍🏻 Would be nice to have before/after screenshots.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte (hunk 1 above)
Comment (vovakulikov) [reply client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte:32]: > You only need to set mac if you want to have a different keyboard shortcut

Are you saying that hotkeysJs will automatically map the option key to alt for Mac (or Windows, basically any OS where we don't have the option as a key?)

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte (hunk 3 above)
Comment (vovakulikov) [reply client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPicker.svelte:16]: Yeah, in the first try it was the command and ctrl keys so I specified them both, will remove them

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte (hunk 1 above)
Comment (vovakulikov) [reply client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/+layout.svelte]: 🤦 my bad, I was thinking about windows key

File: client/web-sveltekit/src/lib/fuzzyfinder/FuzzyFinderContainer.svelte [hunk 6]
```diff
@@ -65,6 +66,16 @@
...
 
+    registerHotkey({
+        keys: { key: 'Esc' },
+        ignoreInputFields: false,
+        handler: event => {
+            event.preventDefault()
+            fuzzyFinderState.update(state => ({ ...state, open: false }))
+            return false
+        },
+    })
```
Comment (fkling) [inline client/web-sveltekit/src/lib/fuzzyfinder/FuzzyFinderContainer.svelte:77]: We should already get this behavior for free via the `<dialog>` component. At least it works for me in Firefox.

Comment (vovakulikov) [reply client/web-sveltekit/src/lib/fuzzyfinder/FuzzyFinderContainer.svelte:77]: Didn't work in my 15.6 Safari for some reason, I'll remove it if it's not needed as soon as we setup our browser list configuration

**Pull request merged**